src/App.jsx

```diff
@@ -18,55 +21,80 @@ import Footer from '@/components/footer/Footer';
 import { ThemeProvider } from '@material-ui/core/styles';
 import theme from '@/theme';
 
-import UserContext from '@/user';
+import UserContext from '@/context/user';
+import AlertContext from '@/context/alert';
 
 export default function App() {
   const [user, setUser] = useState(null);
   const isSignedIn = Boolean(user && user.username);
+
+  const [alert, setAlert] = useState({});
+
+  function simpleSetAlert(severity, msg) {
+    setAlert({ severity, msg });
+  }
+
   return (
     <div id="pageContainer">
-      <UserContext.Provider value={user}>
-        <ThemeProvider theme={theme}>
-          <Header user={user} setUser={setUser} />
-          <div id="contentContainer">
-            <Switch>
-              <Route path="/about">
-                <About />
-              </Route>
-              <Route path="/help">
-                <Help />
-              </Route>
-              <Route path="/guide">
-                <Guide isSignedIn={isSignedIn} />
-              </Route>
-              <Route path="/neighborhood">
-                <Neighborhood />
-              </Route>
-              <Route path="/questions">
-                <QuestionList />
-              </Route>
-              <Route path="/question/:question_id">
-                <QuestionAnswerViewer />
-              </Route>
-              <Route path="/termsofservice">
-                <TermsofService />
-              </Route>
-              <Route
-                path="/simple"
-                render={({ match: { url } }) => (
-                  <>
-                    <Route path={`${url}/view`} component={() => SimpleViewer({ user })} exact />
-                  </>
-                )}
-              />
-              <Route path="/">
-                <Landing isSignedIn={isSignedIn} />
-              </Route>
-            </Switch>
-          </div>
-          <Footer />
-        </ThemeProvider>
-      </UserContext.Provider>
+      <AlertContext.Provider value={simpleSetAlert}>
+        <UserContext.Provider value={user}>
+          <ThemeProvider theme={theme}>
+            <Snackbar
```

Comment:
We can't use a reusable alert component because it would be deleted on navigation. If we display "Question deleted" in an alert component, and then navigate to QuestionList the HTML would be removed from the page. I don't like using a ton of providers but I think it might be the best option here.

Reply:
I'm not sure I understand. Everything outside of the Switch component down below is being rendered on every route, like Header and Footer. How I've done a reusable snackbar before was have a custom hook that contained all the state for the component that was instantiated in the top level App.jsx, and that hook was passed down as a prop to all the components that need it (swap in Provider or whatever will give the children access to it), and also pass it to a 'dumb' snackbar component that displayed whatever it's given. It's similar to what you're doing here, I just would like the snackbar html code in a separate component so our App html doesn't get so cluttered.